perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -6370,6 +6356,132 @@ protected void initialize() throws InternalErrorException {
 		log.debug("AttributesManagerBlImpl initialize ended.");
 	}
 
+	/**
+	 * Finds all attributes that depend on given attribute definition.
+	 * ATTENTION: before using this method, make sure there is no cycle in given dependencies.
+	 *
+	 * @param key attribute definition
+	 * @param inverseDependencies inverse dependencies
+	 * @param inverseStrongDependencies inverse strong dependencies
+	 * @return Set of all attribute definitions that depend on given attribute definition
+	 */
+	private Set<AttributeDefinition> findAllAttributeDependencies(
+			AttributeDefinition key,
+			Map<AttributeDefinition, Set<AttributeDefinition>> inverseDependencies,
+			Map<AttributeDefinition, Set<AttributeDefinition>> inverseStrongDependencies) {
+		Set<AttributeDefinition> dependenciesOfAttribute = new HashSet<>();
+
+		dependenciesOfAttribute.addAll(inverseStrongDependencies.get(key));
+		dependenciesOfAttribute.addAll(inverseDependencies.get(key));
+		List<AttributeDefinition> stackingAttributes = new ArrayList<>(inverseStrongDependencies.get(key));
+		while (!stackingAttributes.isEmpty()) {
+			AttributeDefinition firstAttr = stackingAttributes.get(0);
+			stackingAttributes.remove(firstAttr);
+			dependenciesOfAttribute.addAll(inverseStrongDependencies.get(firstAttr));
+			dependenciesOfAttribute.addAll(inverseDependencies.get(firstAttr));
+			stackingAttributes.addAll(inverseStrongDependencies.get(firstAttr));
+		}
+
+		return dependenciesOfAttribute;
+	}
+
+	/**
+	 * Generates inverse dependencies from given dependencies
+	 *
+	 * @param dependencies input dependencies
+	 * @param inverseDependencies output inversed dependencies
+	 */
+	private void generateInverseDependencies(Map<AttributeDefinition, Set<AttributeDefinition>> dependencies,
+		                                       Map<AttributeDefinition, Set<AttributeDefinition>> inverseDependencies) {
```

Comment:
Please, use inverseDependencies as return type and change the code to process it this way.